### DISCUSSION TITLE
Role based login in ng-matero application

### DISCUSSION THREAD

**Discussion by rahulpd101** [ng-matero/ng-matero#566] [Q&A] [answered]
Hi @nzbin/any other interested person,
I have been trying to implement role based login in your application. I added my credentials in the '**in-mem-data.service.ts**' file with a '_roles_' property set to '_GUEST_' but couldn't help understand as to why it is not working.
Please help me out here. I have also referred to '**ngx-permissions**' library but couldn't find any resource related to _role based_ login.

**Comment by nzbin** [comment, marked as answer]
The `in-mem-data.service` is just used for example, I'm not sure how you does with it. If you have an API with roles, you should call it and bind the roles in the `startup.service`. After that, you can get the roles and permissions with `ngx-permissions` services in the whole app.

https://github.com/ng-matero/ng-matero/blob/6057d69a3f8399420f8a0957d4de3f9387741c2c/src/app/core/bootstrap/startup.service.ts#L43-L52
